src/storage/src/storage_state.rs

```diff
@@ -130,17 +133,16 @@ impl<'w, A: Allocate> Worker<'w, A> {
                             crate::render::build_storage_dataflow(
                                 &mut self.timely_worker,
                                 &mut self.storage_state,
-                                &source.id.to_string(),
-                                source.clone(),
+                                &id.to_string(),
+                                ingestion,
+                                Antichain::from_elem(Timestamp::minimum()),
```

Comment:
Is the reason that we have this argument (that's always hardcoded to `[minimum]` right now) because it will be used to do the optimization where we don't re-ingest all of the source?

Reply:
That's right and we should also rename this to something else like `restored_upper` or `resume_upper` because we're not really asking the ingestion pipeline to produce a snapshot at the `as_of` frontier

Reply:
actually it's a little more complicated. i'll leave it as-is in this PR and keep thinking about it. The tricky bit is that this frontier is treated as `initial_upper` everywhere except where we read back another collection using `persist_source` (this is the case for debezium tx topic right now) and `persist_source` really does have `as_of` semantics. I'll ruminate on it a bit